CMakeLists.txt

```diff
@@ -233,6 +233,15 @@ zephyr_compile_options(${CONFIG_COMPILER_OPT_AS_LIST})
 # TODO: This Clang check is broken
 if(CMAKE_C_COMPILER_ID STREQUAL "Clang")
   zephyr_cc_option(
+    #FIXME: need to fix all of those
+    -Wno-sometimes-uninitialized
+    -Wno-shift-overflow
+    -Wno-missing-braces
+    -Wno-self-assign
+    -Wno-address-of-packed-member
+    -Wno-unused-function
+    -Wno-initializer-overrides
+    -Wno-section
```

Comment:
Would it be simpler to just suppress -Werror when the compiler is clang until these get cleaned up?  I presume they're real warnings we want to fix, so visibility would be better.

Reply:
Assuming clang is/will be in CI it is better to do it like this because then you can prevent more warnings from occurring in the interim of fixing these warnings.

Application developers don't want to see warnings that they did not cause.

Reply:
It would be great to factor out these flags to their respective ${COMPILER}.cmake and toolchain cmake files.